### PR TITLE
#44 - Create a DropDown Component

### LINKED ISSUE #44: Create a dropdown component
Support the interface with properties `label` and `value` for dropdown.
Develop a custom UI for dropdown items.

image

### PR DESCRIPTION
**Pull request by Koushith**
## Description

- [x] Adds media query for Smaller Devices
- [x] Adds Error Message Support
- [x] Removed redundant Options Component.

## Screenshot

image
image

## Platform tested
- [x] Web
- [x] Mobile

### CODE REVIEW

File: src/components/primitive/drop-down/drop-down.component.styles.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+import { DropDwonProps } from './drop-down.component.props';
+
+export const DropdownWrapper = styled.div<DropDwonProps>`
+  display: flex;
+  width: ${({ wide }) => (wide ? '40rem' : '100%')} !important;
+  flex-flow: column;
+  justify-content: flex-start;
+`;
+
+export const StyledSelect = styled.select`
```
Comment (koshikraj) [inline src/components/primitive/drop-down/drop-down.component.styles.ts]: I think we need a border for the select style, color defaulted to primary as the background is plain on all screens. Refer the design.

Comment (Koushith) [reply src/components/primitive/drop-down/drop-down.component.styles.ts]: yeah, added in the latest commit.

File: src/components/primitive/drop-down/drop-down.component.tsx [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+  return (
+    <DropdownWrapper {...rest}>
+      <StyledLabel>{label}</StyledLabel>
+      <StyledSelect>{children}</StyledSelect>
+    </DropdownWrapper>
+  );
+};
+
+export const Option: React.FC<DropDownOptionProps> = (props) => {
```
Comment (dsouzaedison) [inline src/components/primitive/drop-down/drop-down.component.tsx]: Don't export this. Options must be rendered within the dropdown component. It's not obvious to consumers of dropdown component that Option must be imported. Instead, make options[] as a prop.

File: src/components/primitive/drop-down/drop-down.component.props.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
+export interface DropDownComponentProps {
+  /**
+   * Label for the Dropsdown
```
Comment (dsouzaedison) [inline src/components/primitive/drop-down/drop-down.component.props.ts]: typo

**Review by dsouzaedison — CHANGES_REQUESTED** [review]

File: src/components/primitive/drop-down/drop-down.component.tsx (hunk 2 above)
Comment (Koushith) [reply src/components/primitive/drop-down/drop-down.component.tsx]: yeah, removed in the latest commit.

File: src/components/primitive/drop-down/drop-down.component.tsx [hunk 4]
```diff
@@ -1,19 +1,35 @@
...
+  const { label, wide, options, children, error, ...rest } = props;
 
   return (
-    <DropdownWrapper {...rest}>
+    // @ts-ignore - No overload matched this call.
+    <DropdownWrapper wide={wide}>
       <StyledLabel>{label}</StyledLabel>
-      <StyledSelect>{children}</StyledSelect>
+      <StyledSelect error={error} {...rest}>
+        {options !== undefined &&
```
Comment (dsouzaedison) [inline src/components/primitive/drop-down/drop-down.component.tsx]: If I pass null, this code will render? Don't check for a specific case. Always check for nullish values by converting them to boolean.

**Pull request merged**